Formula/petsc-complex.rb

```diff
@@ -44,16 +44,15 @@ def install
     # Avoid references to Homebrew shims
     rm_f lib/"petsc/conf/configure-hash"
 
-    if OS.mac?
-      inreplace lib/"petsc/conf/petscvariables", Superenv.shims_path, ""
-    elsif File.readlines("#{lib}/petsc/conf/petscvariables").grep(Superenv.shims_path.to_s).any?
+    if OS.mac? || File.foreach("#{lib}/petsc/conf/petscvariables").any? { |l| l[Superenv.shims_path.to_s] }
       inreplace lib/"petsc/conf/petscvariables", Superenv.shims_path, ""
```

Comment:
```suggestion
      inreplace lib/"petsc/conf/petscvariables", "#{Superenv.shims_path}/", ""
```
I think this is needed or else you leave paths like `/clang` which clearly won't work.

Reply:
Are we fine with existing bottles or do we want to rebuild/revision-bump?

Reply:
Could probably rebuild bottles. Not sure about `revision` bump. In the absence of complaints I'd suggest skipping it.